contracts/debt-contracts/TermsContract.sol

```diff
@@ -49,86 +57,154 @@ contract TermsContract is ITermsContract {
     constructor(
         address _principalTokenAddr,
         uint256 _principal,
-        uint256 _amortizationUnitType,
-        uint256 _termLength,
-        uint256 _termPayment,
-        uint256 _gracePeriodLength,
-        uint256 _gracePeriodPayment,
+        uint256 _timeUnitType,
+        uint256 _loanPeriod,
         uint256 _interestRate
     ) public {
+        require(_principalTokenAddr != address(0), "Loaned token must be an ERC20 token"); //TODO(Dan): More rigorous way of testing ERC20?
+        require(_timeUnitType < 5, "Invalid time unit type");
+        require(_loanPeriod > 0, "Loan period must be higher than 0");
```

Comment:
@onggunhao @tspoff  will we also want a maximum `loanPeriod`, according to the research, the soft dpt was 1 year